test/integration/authentication_test.go

```diff
@@ -94,327 +79,45 @@ func TestKubernetesAuthentication(t *testing.T) {
+			Username: "alex@acme.com",
+			Groups:   []string{"OnCall", "Engineering"},
+		},
+		gatewayPort,
 		kindKubeConfig.Host,
+		controller.URL,
 	)
-	defer client.Close()
+	if err != nil {
+		t.Fatalf("Failed to create user: %v", err)
+	}
+	defer user.Close()
 
-	clientKubectl := createClientKubectl(t, kindKubectl, client.URL)
+	testutil.CreateK8sRoleBinding(t, kindKubectl, []string{user.Username})
 
 	// Test `kubectl auth whoami`
-	output, err := clientKubectl.Command("auth", "whoami", "-o", "json")
+	output, err := user.Kubectl.Command("auth", "whoami", "-o", "json")
 	if err != nil {
 		t.Fatalf("Failed to execute kubectl auth whoami: %v", err)
 	}
 
-	assertWhoAmI(t, output, user.Username, append(user.Groups, "system:authenticated"))
+	testutil.AssertWhoAmI(t, output, user.Username, append(user.Groups, "system:authenticated"))
 
 	expectedUser := map[string]any{
 		"id":       "user-1",
 		"username": "alex@acme.com",
 		"groups":   []any{"OnCall", "Engineering"},
 	}
-	assertLogsForREST(t, logs, "/apis/authentication.k8s.io/v1/selfsubjectreviews", expectedUser)
+	testutil.AssertLogsForREST(t, logs, "/apis/authentication.k8s.io/v1/selfsubjectreviews", expectedUser)
 
 	// Test `kubectl exec`
-	output, err = clientKubectl.Command("exec", "test-pod", "--", "cat", "/etc/hostname")
+	output, err = user.Kubectl.Command("exec", "test-pod", "--", "cat", "/etc/hostname")
 	if err != nil {
 		t.Fatalf("Failed to execute kubectl exec: %v", err)
 	}
 
 	assert.Equal(t, "test-pod\n", string(output))
-	assertLogsForExec(t, logs, "/api/v1/namespaces/default/pods/test-pod/exec?command=cat&command=%2Fetc%2Fhostname&container=test-pod&stderr=true&stdout=true", "test-pod", expectedUser)
-}
-
-const setupYaml = `
-# Setup default service account for impersonation
-#
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
-metadata:
-  name: gateway-impersonation
-rules:
-- apiGroups:
-  - ""
-  resources:
-  - users
-  - groups
-  - serviceaccounts
-  verbs:
-  - impersonate
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: gateway-impersonation
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: gateway-impersonation
-subjects:
-- kind: ServiceAccount
-  name: default
-  namespace: default
----
-apiVersion: v1
-kind: Secret
-metadata:
-  name: gateway-default-service-account
-  annotations:
-    kubernetes.io/service-account.name: default
-type: kubernetes.io/service-account-token
----
-
-# Setup a test pod
-#
-apiVersion: v1
-kind: Pod
-metadata:
-  name: test-pod
-spec:
-  containers:
-  - name: test-pod
-    image: busybox
-    command: ["sleep", "3600"]
----
-
-# Setup role binding for test user
-#
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: gateway-test-user
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: edit
-subjects:
-- kind: User
-  name: "alex@acme.com"
-  apiGroup: rbac.authorization.k8s.io
-`
```

Comment:
This cluster role binding YAML is extracted to a new method [`testutil.CreateK8sRoleBinding`](https://github.com/Twingate/kubernetes-access-gateway/pull/29/files#diff-eed45dbc0ee3e34a23d4a90d5b9cfb09a74e06a8a034c0f5f6e4c529df4f1b0dR160) so we could bind the `edit` role to more than 1 user.